### DISCUSSION TITLE
How do I hook a function?

### DISCUSSION THREAD

**Discussion by Anixx** [ramensoftware/windhawk#150] [General]
Suppose I want to hook this function, what should I do?

typedef UINT (WINAPI *PrivateExtractIconsW_t)(
    LPCWSTR szFileName,
    int nIconIndex,
    int cxIcon,
    int cyIcon,
    HICON *phicon,
    UINT *piconid,
    UINT nIcons,
    UINT flags);

**Comment by m417z** [comment]
Use `Wh_SetFunctionHook`. Look at the new mod template example or at existing mods. For more help, you can ask questions in [the Discord channel of Windhawk](https://discord.gg/WZgXScMud7).
